src/kem/bike/additional_r4/gf2x_mul_base_portable.c

```diff
@@ -41,14 +41,29 @@ void gf2x_mul_base_port(OUT uint64_t *c,
   u[7] = u[6] ^ b0m;
 
   // Step 2: Multiply two elements in parallel in positions i, i+s
-  l = u[LSB3(a0)] ^ (u[LSB3(a0 >> 3)] << 3);
-  h = (u[LSB3(a0 >> 3)] >> 61);
+  for (size_t i = 0; i < 8; ++i) {
+      // use a mask for secret-independent memory access
+      mask = LSB3(a0) - i;
+      l ^= (u[i] & (uint64_t)(0 - (1 - ((uint64_t)(mask | (0 - mask)) >> 63))));
```

Comment:
maybe we can extract the mask computation to a function in `utilities.h`, something like:
```
// Returns 0xfff..ff if a == b, otherwise returns 0.
uint64_t secure_cmpeq64_mask(uint64_t a, uint64_t b) {
  uint64_t tmp = a - b;
  return (uint64_t)(0 - (1 - ((uint64_t)(tmp | (0 - tmp)) >> 63))));
}
```
and then use it here and in other places below like this:
```
l ^= u[i] & secure_cmpeq64_mask(LSB3(a0), i);
```

Reply:
Done!